src/api/install.ts

```diff
@@ -521,23 +522,31 @@ async function installInContext (
     outdatedPkgs: installCtx.outdatedPkgs,
   })
 
+  ctx.pendingBuilds = ctx.pendingBuilds
+    .filter(pkgId => !result.removedPkgIds.has(dp.resolve(ctx.wantedShrinkwrap.registry, pkgId)))
+
+  if (opts.ignoreScripts) {
+    ctx.pendingBuilds = ctx.pendingBuilds
```

Comment:
In this case duplication might appear `pendingBuilds`. `R.union()` can concat them removing duplicates.